src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/WindowsRuntime/Generated/WinRT/ExceptionHelpers.cs

```diff
@@ -0,0 +1,396 @@
+
+                        restrictedErrorInfo.GetErrorDetails(out string description,
+                                                            out int restrictedErrorInfoHResult,
+                                                            out string restrictedDescription,
+                                                            out string capabilitySid);
+
+                        // Since this is a special case where by convention there may be a correlation, there is not a
+                        // guarantee that the restricted error info does belong to the async error code.  In order to
+                        // reduce the risk that we associate incorrect information with the exception object, we need
+                        // to apply a heuristic where we attempt to match the current exception's HRESULT with the
+                        // HRESULT the IRestrictedErrorInfo belongs to.  If it is a match we will assume association
+                        // for the IAsyncInfo case.
+                        if (e.HResult == restrictedErrorInfoHResult)
+                        {
+                            e.AddExceptionDataForRestrictedErrorInfo(description,
+                                                                    restrictedDescription,
+                                                                     restrictedErrorInfo.GetReference(),
+                                                                     capabilitySid,
+                                                                     restrictedErrorInfoRef.As<ABI.WinRT.Interop.IRestrictedErrorInfo.Vftbl>());
+                        }
+                    }
+                }
+                catch
+                {
+                    // If we can't get the restricted error info, then proceed as if it isn't associated with this
+                    // error.
+                }
+            }
+
+            return e;
+        }
+    }
+
+    internal static class ExceptionExtensions
+    {
+        public static void SetHResult(this Exception ex, int value)
+        {
+            ex.GetType().GetProperty("HResult").SetValue(ex, value);
+        }
+
+        internal static Exception GetExceptionForHR(this Exception innerException, int hresult, string messageResource)
+        {
+            Exception e;
+            if (innerException != null)
+            {
+                string message = innerException.Message ?? messageResource;
+                e = new Exception(message, innerException);
+            }
+            else
+            {
+                e = new Exception(messageResource);
+            }
+            e.SetHResult(hresult);
+            return e;
+        }
+    }
+
+    internal class ErrorStrings
+    {
+        internal static string Format(string format, params object[] args) => String.Format(format, args);
+
+        internal static readonly string Arg_IndexOutOfRangeException = "Index was outside the bounds of the array.";
```

Comment:
Does the code here need to consider multi-language support. Such as Chinese, Japanese and so on

Reply:
This code never considered multi-language support when it used to live in CoreCLR, so I don't believe it needs to here. Someone on the WPF team can correct me if that is not the case.

Reply:
@ryalanms @fabiant3 should decide this, but seeing as this is a temporary change and the original CoreCLR code isn't localized, I agree we shouldn't do anything special here.

Reply:
I don't think this needs multi-lang support right now - as this is temporary/transient. 

In general, WPF localizes exception strings. Consuming an unlocalized exception string would break the norm. 

That said, WPF also links against other DLL's (that can in turn throw their own exceptions) that have different localization norms, so this is no different. 

My 2c. New tools/software like CsWinRT should be designed with globalization needs in mind. Hopefully hard-coding en-us strings is just indicative of the fact that the tool is an early prototype.